tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ComputeGhQuantities.cpp

```diff
@@ -574,4 +579,93 @@ SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.GhQuantities",
                                     upper_bound);
   test_shift_deriv_functions_analytic(solution, grid_size, lower_bound,
                                       upper_bound);
+
+  // Check that compute items work correctly in the DataBox
+  // First, check that the names are correct
+  CHECK(GeneralizedHarmonic::Tags::TimeDerivSpatialMetricCompute<
+            3, Frame::Inertial>::name() == "dt(SpatialMetric)");
+  CHECK(GeneralizedHarmonic::Tags::TimeDerivLapseCompute<
+            3, Frame::Inertial>::name() == "dt(Lapse)");
+  CHECK(GeneralizedHarmonic::Tags::TimeDerivShiftCompute<
+            3, Frame::Inertial>::name() == "dt(Shift)");
+
+  // Check that the compute items return the correct values
+  const DataVector used_for_size{3., 4., 5.};
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> distribution(-0.1, 0.1);
+
+  const auto& spatial_metric =
```

Comment:
none of these are references... They're just `const auto`s

Reply:
I believe declaring them as `const references` binds them to the temporary objects created by `make_with_random_values`. Declaring them `const auto` would cause the contents of those temporary objects to be copied over into new memory. So, this saves a copy op.

Reply:
No, it doesn't change anything other than lie to the user that these are references when they're not. Here's the assembly: https://godbolt.org/z/5bSuTr. Both in debug and release the same code is generated so we should write the code to express what it actually is doing rather than something else, which is why I don't think they should be references.

Reply:
It appears that when I turn down compiler optimization level to `-O1`, I do begin to see differences in assembly: https://godbolt.org/z/rjFqz9. An address resolving `lea` op is replaced by `mov` when going from `const auto&` to `const auto`. However at any higher optimization levels, the compiler is optimizing these difference away.

> Both in debug and release the same code is generated so we should write the code to express what it actually is doing rather than something else, which is why I don't think they should be references.

Well, is it clear that the machine-code that the compiler is generating corresponds to initialize-by-value? I think the generated code corresponds to initialize-as-const-ref, although I could be grossly mistaken.

Reply:
I'm not sure what your argument here is. My argument is that `const auto` is correct and easier to understand. I don't see any reason to write more confusing code in a test to get performance that you don't actually get

Reply:
I'm not arguing to keep my code as is, just trying to understand what the right thing to do here is and why. Ttyp

Reply:
:+1: I think `const auto` is right when you are binding to a member inside a class (e.g. return things from the DataBox) because:
1. it makes it clear that you are taking ownership of the object 
2. Nothing else can mess with the object "behind your back"
3. With optimizations and copy elision there is no performance hit.

My main reason really is that I think it's confusing to pretend something is a "pointer" when it's not :)

Reply:
Okay, I'm convinced. I've changed `const auto&` --> `const auto` in this and other related PRs